src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,561 @@
+    re_step_down = re.compile(r"rdb_raft_step_down.*([0-9a-fA-F]{8}).*leader of term (\d+)")
+
+    # Future possibility: target state change events
+    # update_one_tgt(), update_tgt_down_drain_to_downout()
+    #   "change Target.*rank (\d+) idx (\d+).*to (\w+)"
+    #   need those functions to print pool UUID. Then here, store a list of target change events in
+    #   tgt_change = {"rank": affected_rank, "tgt_idx": affected_tgtidx, "state": new_tgtstate}
+    #   self._pools[puuid][term]["maps"][mapver]["tgt_state_changes"].append(tgt_change)
+
+    # pool map version update events
+    upd_re = r"ds_pool_tgt_map_update.*([0-9a-fA-F]{8}): updated.*map: version=(\d+)->(\d+) pointer"
+    re_pmap_update = re.compile(upd_re)
+
+    # uniform rebuild string identifier rb=<pool_uuid>/<rebuild_ver>/<rebuild_gen>/<opcode_string>
+    rbid_re = r"rb=([0-9a-fA-F]{8})/(\d+)/(\d+)/(\w+)"
+
+    # Future possibility: match the rebuild preliminary steps
+    # rebuild_task_ult() wait for scheduling, and map dist - both would info to match on.
+    # re.compile(r"rebuild_task_ult\(\).*rebuild task sleep (\d+) second")
+    # re.compile(r"rebuild_task_ult\(\).*map_dist_ver (\d+) map ver (\d+)")
+
+    # Rebuild: PS leader engine starting and status checking a given operation
+    # statuses: "scanning", "pulling", "completed", "aborted"
+    ldr_start_re = "rebuild_leader_start.*" + rbid_re + "$"
+    ldr_status_re = r"rebuild_leader_status_check\(\).*" + rbid_re + r" \[(\w+)\].*duration=(\d+)"
+    re_rebuild_ldr_start = re.compile(ldr_start_re)
+    re_rebuild_ldr_status = re.compile(ldr_status_re)
+
+    # Legacy rebuild PS leader logging (before uniform rebuild string)
+    old_ldr_start_re = r"rebuild_leader_start.*([0-9a-fA-F]{8}).*version=(\d+)/(\d+).*op=(\w+)"
+    old_ldr_status_re = (r"rebuild_leader_status_check\(\) (\w+) \[(\w+)\] \(pool ([0-9a-fA-F]{8}) "
+                         r"leader (\d+) term (\d+).*ver=(\d+),gen (\d+).*duration=(\d+) secs")
+    re_old_ldr_start = re.compile(old_ldr_start_re)
+    re_old_ldr_status = re.compile(old_ldr_status_re)
+
+    def __init__(self):
+        # dictionaries indexed by pool UUID
+        self._pools = {}
+        self._highest_pmapver = {}
+        self._cur_ldr_rank = {}
+        self._cur_ldr_pid = {}
+        self._cur_term = {}
+
+        # filename to rank map
+        self._file_to_rank = {}
+
+        self._warnings = []
+        self._check_rb_new_fmt = True
+        self._check_rb_legacy_fmt = True
+        self._debug = False
+
+    def _warn(self, wmsg, fname, line=None):
+        full_msg = f"WARN file={fname}"
+        if line:
+            full_msg += f", line={line.lineno}"
+        full_msg += f": {wmsg}"
+
+        self._warnings.append(full_msg)
+        print(full_msg)
+
+    def get_warnings(self):
+        """Return all warnings stored when scanning engine log files"""
+        return self._warnings
```

Comment:
You could make this a property
```suggestion
    @property
    def warnings(self):
        """Return all warnings stored when scanning engine log files"""
        return self._warnings
```
And instead of
```
obj.get_warnings()
```
It would be like
```
obj.warnings  # no parens
```

Reply:
done in  latest commit